plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardJre7ProjectFacetDetector.java

```diff
@@ -0,0 +1,97 @@
+import java.util.logging.Logger;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.jst.common.project.facet.core.JavaFacet;
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
+import org.eclipse.wst.common.project.facet.core.IFacetedProjectWorkingCopy;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetDetector;
+import org.xml.sax.SAXException;
+
+public class AppEngineStandardJre7ProjectFacetDetector extends ProjectFacetDetector {
+  private static final Logger logger =
+      Logger.getLogger(AppEngineStandardJre7ProjectFacetDetector.class.getName());
+
+  @Override
+  public void detect(IFacetedProjectWorkingCopy workingCopy, IProgressMonitor monitor)
+      throws CoreException {
+    SubMonitor progress = SubMonitor.convert(monitor, 5);
+    String projectName = workingCopy.getProjectName();
+
+    IFile appEngineWebXml =
+        WebProjectUtil.findInWebInf(workingCopy.getProject(), new Path("appengine-web.xml"));
+    progress.worked(1);
+    if (appEngineWebXml == null || !appEngineWebXml.exists()) {
+      logger.fine("skipping " + projectName + ": no appengine-web.xml found");
+      return;
+    }
+    try (InputStream content = appEngineWebXml.getContents()) {
+      AppEngineDescriptor descriptor = AppEngineDescriptor.parse(content);
+      progress.worked(1);
+      if (descriptor.getRuntime() != null && !"java7".equals(descriptor.getRuntime())) {
+        logger.fine("skipping " + projectName + ": appengine-web.xml is not java7");
+        return;
+      }
+      logger.fine(projectName + ": appengine-web.xml has java7 runtime");
+      
+      logger.fine(projectName + ": setting App Engine Standard JRE7 facet");
+      workingCopy.addProjectFacet(AppEngineStandardFacet.JRE7);
+      progress.worked(1);
+
+      if (!workingCopy.hasProjectFacet(JavaFacet.VERSION_1_7)) {
+        if (workingCopy.hasProjectFacet(JavaFacet.FACET)) {
+          logger.fine(projectName + ": changing to Java 7 facet");
+          workingCopy.changeProjectFacetVersion(JavaFacet.VERSION_1_7);
+        } else {
+          logger.fine(projectName + ": setting Java 7 facet");
+          Object javaModel = FacetUtil.createJavaDataModel(workingCopy.getProject());
+          workingCopy.addProjectFacet(JavaFacet.VERSION_1_7);
+          workingCopy.setProjectFacetActionConfig(JavaFacet.FACET, javaModel);
+        }
+        progress.worked(1);
+      }
+
+      if (!workingCopy.hasProjectFacet(WebFacetUtils.WEB_25)) {
+        if (workingCopy.hasProjectFacet(WebFacetUtils.WEB_FACET)) {
+          logger.fine(projectName + ": changing to Dynamic Web 2.5 facet");
+          workingCopy.changeProjectFacetVersion(WebFacetUtils.WEB_25);
+        } else {
+          logger.fine(projectName + ": setting Dynamic Web 2.5 facet");
```

Comment:
I wasn't concerned about the logging level. Rather it was of an aesthetic reason and about the code taking up screen real estate. I don't really care, so leave them if you want.